components/drafts/panel/panel_body.tsx

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback} from 'react';
+import {useDispatch} from 'react-redux';
+
+import {removeFilePreview} from 'actions/views/drafts';
+
+import {UserProfile, UserStatus} from 'mattermost-redux/types/users';
+
+import {PostDraft} from 'types/store/rhs';
+
+import Markdown from 'components/markdown';
+import FilePreview from 'components/file_preview';
+import ProfilePicture from 'components/profile_picture';
+import {imageURLForUser} from 'utils/utils';
+
+import './panel_body.scss';
+
+type Props = {
+    channelId: string;
+    displayName: string;
+    draftId: string;
+    fileInfos: PostDraft['fileInfos'];
+    message: string;
+    status: UserStatus['status'];
+    uploadsInProgress: PostDraft['uploadsInProgress'];
+    user: UserProfile;
+}
+
+function Body({
+    channelId,
+    displayName,
+    draftId,
+    fileInfos,
+    message,
+    status,
+    uploadsInProgress,
+    user,
+}: Props) {
+    const dispatch = useDispatch();
+
+    const handleRemovePreview = useCallback((id: string) => {
+        dispatch(removeFilePreview(draftId, id));
+    }, [draftId]);
+
+    return (
+        <div className='DraftPanelBody post'>
+            <div className='DraftPanelBody__left post__img'>
+                <ProfilePicture
+                    status={status}
+                    channelId={channelId}
+                    username={user.username}
+                    userId={user.id}
+                    size={'md'}
+                    src={imageURLForUser(user.id)}
+                />
+            </div>
+            <div className='post__content'>
+                <div className='DraftPanelBody__right'>
+                    <div className='post__header'>
+                        <strong>{displayName}</strong>
+                    </div>
+                    <div className='post__body'>
+                        <Markdown message={message}/>
+                    </div>
+                    {(fileInfos.length > 0 || uploadsInProgress?.length > 0) && (
+                        <FilePreview
+                            fileInfos={fileInfos}
```

Comment:
This might be a bit of an overkill, but @hmhealey what do you think about having a util function like: `arrayOrEmpty<T>(in: T[]): T[]` that if the array is empty, returns a constant object with the empty list? That way we may (or may not) reduce some re-renderings here and there.

Reply:
Here might not be a problem, but in many index.ts we have the case of `return x || []`, and try to improve that by passing a "constant empty list". A function like that could help us to avoid extra code around those parts.

Reply:
In those cases, I'd prefer to just pass around the null/undefined value and handle it when the variable is used. It does require null checking, but TS should ensure we do that properly. It can be nice to just pass around an empty array since it works for looping over and such, but we also do that for empty objects sometimes which causes a lot of problems with type checking and how we determine whether or not an object is empty or not.